src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +41,161 @@ const ActorInit Dm_Hina_InitVars = {
+        this->actor.world.pos.y += 40.0f;
+        this->actionFunc = DmHina_AwaitMessageBoxClosing;
+    }
+}
+
+void DmHina_AwaitMessageBoxClosing(DmHina* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+        this->timer = 2;
+        this->actionFunc = DmHina_SetupSubCamera;
+    }
+}
+
+void DmHina_SetupSubCamera(DmHina* this, PlayState* play) {
+    this->timer--;
+    if (this->timer == 0) {
+        this->isVisible = true;
+        Cutscene_Start(play, &play->csCtx);
+        this->subCamId = Play_CreateSubCamera(play);
+        Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STATUS_WAIT);
+        Play_ChangeCameraStatus(play, this->subCamId, CAM_STATUS_ACTIVE);
+        this->actionFunc = DmHina_MoveSubCamera;
+    }
+}
+
+void DmHina_MoveSubCamera(DmHina* this, PlayState* play) {
+    this->subCamEye.x = this->actor.world.pos.x + 100.0f;
+    this->subCamEye.y = this->blueWarpPosY + 260.0f;
+    this->subCamEye.z = this->actor.world.pos.z + 100.0f;
+    this->subCamAt.x = this->actor.world.pos.x;
+    this->subCamAt.y =
+        this->actor.world.pos.y + (this->oscilationFactor * this->yDimScaleFactor) + (40.0f * this->yDimScaleFactor);
+    this->subCamAt.z = this->actor.world.pos.z;
+    Play_SetCameraAtEye(play, this->subCamId, &this->subCamAt, &this->subCamEye);
+    Math_SmoothStepToF(&this->actor.world.pos.y, this->blueWarpPosY + 300.0f, 0.5f, 2.0f, 0.1f);
+    if (((this->blueWarpPosY + 240.0f) < this->actor.world.pos.y) && (this->csState != LIGHT_ORB_STATE_ON)) {
+        this->csState = LIGHT_ORB_STATE_ON;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_WHITE_OUT_INTO_KYOJIN);
+    }
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_MASK_RISING - SFX_FLAG);
+}
+
+void DmHina_SetupLightOrb(DmHina* this, PlayState* play) {
+    s32 i;
+    s16 light1Color;
+
+    switch (this->csState) {
+        case LIGHT_ORB_STATE_OFF:
+            this->lightOrbAlpha = 0;
+            this->scale = 0.0f;
+            break;
+
+        case LIGHT_ORB_STATE_ON:
+            Math_SmoothStepToF(&this->scale, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->lightOrbAlpha = this->scale * 255.0f;
+
+            this->drawOrbEffect = Math_SinS(play->state.frames * 8000);
+
+            for (i = 0; i < 3; i++) {
+                light1Color = this->scale * -255.0f;
+                play->envCtx.lightSettings.diffuseColor1[i] = light1Color;
+                play->envCtx.lightSettings.fogColor[i] = light1Color;
+                play->envCtx.lightSettings.ambientColor[i] = light1Color;
+            }
+
+            play->envCtx.lightSettings.fogNear = this->scale * -500.0f;
+
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.curRoom.segment = NULL;
+            }
+
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    DmHina_SetupLightOrb(this, play);
```

Comment:
```suggestion
    DmHina_UpdateLightOrb(this, play);
```
Seems appropriate based on usage (called in update and updates the light orb)